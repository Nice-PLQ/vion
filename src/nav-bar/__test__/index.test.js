import { mount } from '@vue/test-utils';
import NavBar from '../nav-bar.vue';

describe('NavBar', () => {
  test('渲染', async () => {
    const wrapper = mount(NavBar, {
      props: {
        leftIcon: 'close',
        leftText: '返回',
        rightIcon: 'share',
        rightText: '设置',
        theme: 'black'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes()).toContain('v-navbar_black');
    await wrapper.find('.v-navbar__back').trigger('click');
    await wrapper.find('.v-navbar__left').trigger('click');
    await wrapper.find('.v-navbar__title').trigger('click');
    await wrapper.find('.v-navbar__right').trigger('click');
    await wrapper.find('.v-navbar__right-text').trigger('click');
    expect(wrapper.emitted('back')).toBeTruthy();
    expect(wrapper.emitted('left-click')).toBeTruthy();
    expect(wrapper.emitted('title-click')).toBeTruthy();
    expect(wrapper.emitted('right-click')).toBeTruthy();
    expect(wrapper.emitted('right-text-click')).toBeTruthy();
  });

  test('slots', () => {
    const wrapper = mount(NavBar, {
      slots: {
        left: '<span>left</span>',
        right: '<span>right</span>',
        title: '<span>title</span>'
      },
      props: {
        leftIcon: 'close',
        leftText: '返回',
        rightIcon: 'share',
        rightText: '设置'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
