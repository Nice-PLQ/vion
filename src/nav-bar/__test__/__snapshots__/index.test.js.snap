// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar slots 1`] = `
<div
  class="v-navbar v-navbar_white"
  righticon="share"
>
  <div
    class="v-navbar__status"
  />
  <div
    class="v-navbar__wrap"
  >
    <div
      class="v-navbar__left"
    >
      
      <span>
        left
      </span>
      
    </div>
    <div
      class="v-navbar__title"
    >
      
      <span>
        title
      </span>
      
    </div>
    <div
      class="v-navbar__right"
    >
      
      <span>
        right
      </span>
      
    </div>
  </div>
</div>
`;

exports[`NavBar 渲染 1`] = `
<div
  class="v-navbar v-navbar_black"
  righticon="share"
>
  <div
    class="v-navbar__status"
  />
  <div
    class="v-navbar__wrap"
  >
    <div
      class="v-navbar__left"
    >
      
      <button
        class="v-navbar__back"
        title="返回"
      >
        <i
          class="v-icon"
          style="display: inline-block; font-size: 24px; width: 1em; height: 1em; line-height: 1em;"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M37.4767 13.435L34.6482 10.6066L24.0416 21.2132L13.435 10.6066L10.6066 13.435L21.2132 24.0416L10.6066 34.6482L13.435 37.4767L24.0416 26.8701L34.6482 37.4767L37.4767 34.6482L26.8701 24.0416L37.4767 13.435Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </i>
        <span
          class="v-navbar__back-text"
        >
          返回
        </span>
      </button>
      
    </div>
    <div
      class="v-navbar__title"
    >
      
      
      
      
    </div>
    <div
      class="v-navbar__right"
    >
      
      <button
        class="v-navbar__right-btn"
      >
        <span
          class="v-navbar__right-text"
        >
          设置
        </span>
      </button>
      
    </div>
  </div>
</div>
`;
